pyfive/h5d.py

```diff
@@ -261,8 +264,9 @@ def _get_direct_from_contiguous(self, args=None):
         num_bytes = num_elements*itemsize
        
         # we need it all, let's get it all (i.e. this really does read the lot)
-        self._fh.seek(self.data_offset)
-        chunk_buffer = self._fh.read(num_bytes) 
+        fh = self._fh
```

Comment:
This seems to be the key. I admit to not really understanding why it is working.  Is it because this copy statement results in a different `fh` in each thread? But why? This doesn't obviously do a deepcopy it's just a reference. What am I missing?

Reply:
I don't get it still. I know there wasn't a copy, but why is that statement returning a _new_ file handle, and not returning a reference to the existing one? What do you mean by _without caching it_, and why is this special behaviour?  I understand the distinction between the posix and non-posix route through the code, but I don't understand what is different about doing `self._fh.seek` whether or `fh.seek` if we have just done `fh=self._fh`.